src/components/menu/menu.svelte

```diff
@@ -0,0 +1,234 @@
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    return e.getAttribute('value') ?? e.textContent
+  }
+
+  $: menuItems = Array.from(
+    (
+      popup?.querySelector('.leo-menu-popup slot') as HTMLSlotElement
+    )?.assignedElements() ??
+      popup?.querySelectorAll('.leo-menu-popup > *') ??
+      []
+  )
+
+  $: {
+    for (const [menuItem, index] of menuItems.map((o, i) => [o, i] as const)) {
+      menuItem.setAttribute('tabindex', (index + 1).toString())
+
+      if (menuItem.tagName === 'LEO-OPTION') {
+        if (currentValue === getValue(menuItem)) {
+          menuItem.setAttribute('aria-selected', '')
+        } else menuItem.removeAttribute('aria-selected')
+      } else {
+        menuItem.setAttribute('role', 'menuitem')
+      }
+    }
+  }
+
+  // Note: we check isOpen !== undefined here so this is recalculated every time the
+  // dropdown is opened/closed.
+  $: minWidth =
+    (isOpen !== undefined && target?.getBoundingClientRect().width) || 0
+
+  function selectMenuItem(e: Event) {
+    // Find the option which was clicked on, if any.
+    const item = menuItems.find((item) => e.composedPath().includes(item))
+
+    // If the event was triggered for something which isn't an option don't fire
+    // a change event.
+    if (!item) return
+
+    if (item.tagName === 'LEO-OPTION') {
+      currentValue = getValue(item)
+    }
+
+    isOpen = false
+
+    dispatch('select-item', {
+      value: currentValue
+    })
+
+    dispatch('close')
+  }
+
+  /**
+   * Handles changing the currently focused menu element with the up/down arrow.
+   * @param e The KeyboardEvent
+   */
+  function changeSelection(e: KeyboardEvent) {
+    if (!isOpen || !popup) return
+
+    // Handle closing keys
+    if (e.code === 'Escape') {
+      isOpen = false
+      dispatch('close')
+      return
+    }
+
+    // We allow the user to select options with ArrowUp/ArrowDown as well as
+    // tab/shift+tab.
+    let dir = 0
+    if (e.code == 'ArrowUp') dir -= 1
+    if (e.code === 'ArrowDown') dir += 1
+    if (dir === 0) return
+
+    // First, find the currently focusedIndex. If no option is selected, we'll
+    // select the first option. Otherwise, we select the next/previous item (and
+    // wrap around).
+    let focusedIndex = menuItems.findIndex((e) => e.matches(':focus-within'))
+    if (focusedIndex === -1) {
+      focusedIndex = 0
+    } else {
+      focusedIndex += dir
+      if (focusedIndex < 0) focusedIndex = menuItems.length - 1
+      if (focusedIndex >= menuItems.length) focusedIndex = 0
+    }
+
+    ;(menuItems[focusedIndex] as any)?.focus()
+    e.preventDefault() // preventDefault, so we don't accidentally scroll
+  }
+
+  function handleBlur(e) {
+    isOpen = false
+    dispatch('close')
+  }
+</script>
+
+<div class="leo-menu">
+  {#if isOpen}
+    <Floating {target} placement="bottom-start">
```

Comment:
Nit: add `autoUpdate` prop - this means if the user scrolls around the menu will do it's best to stay in view, while remaining anchored

Reply:
done